support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/authenticator/OAuth20RefreshTokenAuthenticator.java

```diff
@@ -0,0 +1,79 @@
+package org.apereo.cas.support.oauth.authenticator;
+
+import org.apereo.cas.audit.AuditableExecution;
+import org.apereo.cas.authentication.principal.PrincipalResolver;
+import org.apereo.cas.authentication.principal.ServiceFactory;
+import org.apereo.cas.services.ServicesManager;
+import org.apereo.cas.support.oauth.OAuth20Constants;
+import org.apereo.cas.support.oauth.OAuth20GrantTypes;
+import org.apereo.cas.support.oauth.services.OAuthRegisteredService;
+import org.apereo.cas.support.oauth.util.OAuth20Utils;
+import org.apereo.cas.ticket.refreshtoken.OAuth20RefreshToken;
+import org.apereo.cas.ticket.registry.TicketRegistry;
+import org.apereo.cas.util.crypto.CipherExecutor;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.UsernamePasswordCredentials;
+import org.pac4j.core.exception.CredentialsException;
+
+import java.io.Serializable;
+
+/**
+ * This is {@link OAuth20RefreshTokenAuthenticator}.
+ *
+ * @author Julien Huon
+ * @since 6.2.0
+ */
+@Slf4j
+public class OAuth20RefreshTokenAuthenticator extends OAuth20ClientIdClientSecretAuthenticator {
+
+    public OAuth20RefreshTokenAuthenticator(final ServicesManager servicesManager,
+                                            final ServiceFactory webApplicationServiceFactory,
+                                            final AuditableExecution registeredServiceAccessStrategyEnforcer,
+                                            final TicketRegistry ticketRegistry,
+                                            final CipherExecutor<Serializable, String> registeredServiceCipherExecutor,
+                                            final PrincipalResolver principalResolver) {
+        super(servicesManager, webApplicationServiceFactory, registeredServiceAccessStrategyEnforcer,
+            registeredServiceCipherExecutor, ticketRegistry, principalResolver);
+    }
+
+    /**
+     * RefreshTokenAuthenticator can only be used for a refresh token request of a "public" client.
+     * An OAuth "public" client is one that does not define a secret like a mobile application.
+     *
+     * @param context the context
+     * @return true if authenticator can validate credentials.
+     */
+    @Override
```

Comment:
Javadoc support for @Override methods is not strictly necessary. If you do need to explain this, you should do so at the class level.